### PR TITLE
template `DQM/SiPixelPhase1Heterogenous` class to be able to handle the Phase-2 SoA objects

### PR DESCRIPTION
**Pull request by mmusich**
#### PR description:

This is a follow-up to https://github.com/cms-sw/cmssw/pull/39771 and https://github.com/cms-sw/cmssw/pull/38761 in order to be able to fully monitor the SoA products for pixel rechits, tracks and vertices for both the phase-1 and phase-2 upgrade detectors.
As promised in #39771 I also deliver the renaming of all classes and packages from `SiPixelPhase1Heterogeneous` to `SiPixelHeterogeneous` in commit https://github.com/cms-sw/cmssw/pull/40264/commits/64ac830984e24840bffe82618eb795c20a59098c

#### PR validation:

Run successfully 

```
runTheMatrix.py --what upgrade -l 20834.507 -t 4 -j 8
```

and obtained plots for RecHits, Tracks and Vertices.

| RecHits | Tracks | Vertices |
| ------------- | ------------- | ------------- |
| Screenshot from 2022-12-08 20-37-13  | Screenshot from 2022-12-08 20-36-37 | Screenshot from 2022-12-08 20-40-31 |

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

FYI
@AdrianoDee @sroychow

### CODE REVIEW

**Comment by sroychow** [comment]
enable gpu

**Comment by sroychow** [comment]
please test

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 1]
```diff
@@ -239,6 +239,15 @@ def customizeHLTfor38761(process):
 
      return process
 
+def customizeHLTfor40264(process):
+    for producer in producers_by_type(process, "SiPixelPhase1MonitorVertexSoA"):
+        producer._TypedParameterizable__type = "SiPixelMonitorVertexSoA"
```
Comment (mmusich) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:244]: this feels hacky... Is there any better option to rename on the fly?

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
@cmsbuild, please test

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py (hunk 1 above)
Comment (mmusich) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py:244]: Thanks Marino @missirol for the suggestion. I have implemented it in https://github.com/cms-sw/cmssw/pull/40264/commits/dfe16cb1cedd2d9c04d51688ebf2e8cbcc0df761.
I expect that - eventually - after the alpaka migration is completed, this DQM plugins could be removed from the HLT menu itself and then run on DQM machines as part of a standalone client (as for the other subsystems). But that's likely for after January, so let's go with the way you propose.

**Comment by mmusich** [comment]
@cmsbuild, please test

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py (hunk 1 above)
Comment (missirol) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py:244]: Thanks, sounds good.

Fwiw, playing around with the python customisation, this is arguably less hacky (but I'm not asking to make further changes):
```py
process.hltXYZ = cms.EDProducer('NewName', **process.hltXYZ.parameters_())
```

**Comment by emanueleusai** [comment]
+1

- dqm comparisons in failures in CPU appear to be spurious.

**Comment by perrotta** [comment]
+1

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py (hunk 1 above)
Comment (mmusich) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py:244]: Does the last point in https://its.cern.ch/jira/browse/CMSHLT-2603 imply we can go ahead an remove

https://github.com/cms-sw/cmssw/blob/689324234dd4e6a633578b1c39429371f08297dc/DQM/SiPixelHeterogeneous/plugins/SiPixelCompareVertexSoA.cc#L187-L188

and

https://github.com/cms-sw/cmssw/blob/a7263f73bf9a948a9e2fb997a4ccb32c3a548ba4/DQM/SiPixelHeterogeneous/plugins/SiPixelMonitorVertexSoA.cc#L133-L134

or will it be done together with an update of the HLT menu?

Comment (missirol) [reply HLTrigger/Configuration/python/customizeHLTforCMSSW.py:244]: The HLT menus first need to be updated in the release (will be done tomorrow). The duplicate plugins can then removed (I would prefer that's done in a separate PR).

**Pull request merged**